### PR TITLE
Allow to inspect reward accounts wrt #46

### PR DESCRIPTION
**Pull request by hasufell**
Run:
- [x] tests
- [x] stylish
- [x] hlint

Built:
- [x] cabal
- [x] stack
- [x] nix

### CODE REVIEW

File: core/lib/Cardano/Address/Style/Shelley.hs [hunk 1]
```diff
@@ -507,6 +507,23 @@ inspectShelleyAddress addr
...
+                        , "stake_key_hash"    .= base16 (BS.take pubkeyHashSize rest)
+                        , "network_tag"       .= network
+                        ]
+               -- 1111: reward account: scripthash28
+                0b11110000 | BS.length rest == scriptHashSize ->
+                    Just $ Json.object
+                        [ "address_style"     .= Json.String "Shelley"
+                        , "stake_reference"   .= Json.String "by value"
+                        , "stake_key_hash"    .= base16 (BS.take scriptHashSize rest)
+                        , "script_hash"       .= base16 (BS.take scriptHashSize rest)
```
Comment (KtorZ) [inline core/lib/Cardano/Address/Style/Shelley.hs:524]: In the case of stake address, there is either a script hash or a public stake key hash, there can't be both.

Comment (hasufell) [reply core/lib/Cardano/Address/Style/Shelley.hs:524]: I'm still struggling to understand what those all are, even after reading the design spec :hot_face:

**Comment by hasufell** [comment]
> The PR should come with additional tests cases covering this two new cases. See existing scenarios in: https://github.com/input-output-hk/cardano-addresses/blob/master/command-line/test/Command/Address/InspectSpec.hs#L17

done

**Comment by hasufell** [comment]
To consider: allowing to generate these addresses via cardano-addresses.

File: command-line/test/Command/Address/InspectSpec.hs [hunk 2]
```diff
@@ -39,6 +39,14 @@ spec = describeCmd [ "address", "inspect" ] $ do
     specInspectAddress ["Shelley", "by pointer"]
         "addr1gw2fxv2umyhttkxyxp8x0dlpdt3k6cwng5pxj3jhsydzer5ph3wczvf2x4v58t"
 
+    -- reward account: keyhash28
+    specInspectAddress ["Shelley", "by value"]
+        "addr1upshvetj09hxjcm9v9jxgunjv4ehxmr0d3hkcmmvdakx7mq726p32"
+
+    -- reward account: scripthash32
+    specInspectAddress ["Shelley", "by value"]
+        "addr17pshvetj09hxjcm9v9jxgunjv4ehxmr0d3hkcmmvdakx7mrgdp5xsnxe5e6"
```
Comment (KtorZ) [inline command-line/test/Command/Address/InspectSpec.hs:48]: bech32 encoding is only a presentation layer so this works just fine. Yet in principle, stake addresses should use the prefix `stake` to distinguish them from standard addresses. We have started to work on a list of agreed prefixes to facilitate interop between the various tools we have, so have a quick look at: https://github.com/cardano-foundation/CIPs/tree/master/CIP5

File: command-line/test/Command/Address/InspectSpec.hs [hunk 3]
```diff
@@ -39,6 +39,14 @@ spec = describeCmd [ "address", "inspect" ] $ do
     specInspectAddress ["Shelley", "by pointer"]
         "addr1gw2fxv2umyhttkxyxp8x0dlpdt3k6cwng5pxj3jhsydzer5ph3wczvf2x4v58t"
 
+    -- reward account: keyhash28
+    specInspectAddress ["Shelley", "by value"]
+        "addr1upshvetj09hxjcm9v9jxgunjv4ehxmr0d3hkcmmvdakx7mq726p32"
+
+    -- reward account: scripthash32
+    specInspectAddress ["Shelley", "by value"]
```
Comment (KtorZ) [inline command-line/test/Command/Address/InspectSpec.hs:47]: Note that you could make the command superfluous and the test stronger by requiring `"script_hash"` to be present in the output (and respectively, "stake_key_hash") for the above.

**Pull request merged**